homeassistant/components/aprs/device_tracker.py

```diff
@@ -179,5 +180,6 @@ def rx_msg(self, msg: dict):
             for attr in [ATTR_ALTITUDE, ATTR_COMMENT, ATTR_COURSE, ATTR_SPEED]:
                 if attr in msg:
                     attrs[attr] = msg[attr]
+                    attrs["last_beacon"] = dt_util.utcnow()
```

Comment:
This is inside an if statement inside a for loop; this may not accomplish what you're hoping to achieve.

Reply:
Oh you have a good point there. It did accomplish what I was hoping to achieve but very much not in the best way.

I have added a check to see if it is within the "msg" but since that is not a field contained in the APRS packet it should never hit that code correct? Is it still worth adding?

```
attrs = {}
if ATTR_POS_AMBIGUITY in msg:
    pos_amb = msg[ATTR_POS_AMBIGUITY]
    try:
        attrs[ATTR_GPS_ACCURACY] = gps_accuracy((lat, lon), pos_amb)
    except ValueError:
        _LOGGER.warning(
            "APRS message contained invalid posambiguity: %s", str(pos_amb)
        )
if ATTR_LAST_BEACON in msg:
    last_beacon = msg[ATTR_LAST_BEACON]
    try:
        attrs[ATTR_LAST_BEACON] = dt_util.parse_datetime(last_beacon)
    except ValueError:
        _LOGGER.warning(
            "APRS message contained invalid last_beacon timestamp: %s", str(last_beacon)
        )
        """Set the last_beacon to now if invalid dt_util parse."""
        attrs[ATTR_LAST_BEACON] = dt_util.utcnow()
else:
    """APRS message did not contain a last_beacon field."""
    attrs[ATTR_LAST_BEACON] = dt_util.utcnow()

for attr in [ATTR_ALTITUDE, ATTR_COMMENT, ATTR_COURSE, ATTR_SPEED]:
    if attr in msg:
        attrs[attr] = msg[attr]
```

Reply:
It looks like the APRS packet may contain a `timestamp` field, if you're interested in using that.

Reply:
@PhilRW I have made some more modifications and yes you are right the timestamp field is present, I must have missed that. I have added a parser for that if it exists, then if it fails or is missing it will take the current time. 

I have also added 3 test cases, normal (timestamp included in msg), missing (no timestamp in msg), and invalid (bad formatted timestamp in msg).

Do you have any other suggestions?